File: aparcamiento/IvanRC-aparcamiento/IvanRC-aparcamiento-Java/src/main/java/ModelsAparcamiento/Conductor.java
package ModelsAparcamiento;

import java.util.Objects;
import java.util.Scanner;

public class Conductor {

    public static Scanner sc = new Scanner(System.in);

    private static int id = 0;
    private String nombre;
    private String dni;

    private int nextId(){
        id = id + 1;
        return id;
    }


    public Conductor(String nombre, String dni){
        this.id = nextId();
        this.nombre = nombre;
        this.dni = dni;
    }

    public static int getId() {
        return id;
    }

    public String getNombre() {
        return nombre;
    }

    public String getDni() {
        return dni;
    }

    public static void setId(int id) {
        Conductor.id = id;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public void setDni(String dni) {
        this.dni = dni;
    }

    @Override
    public String toString() {
        return "Conductor{" +
                "nombre='" + nombre + '\'' +
                ", dni='" + dni + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Conductor conductor = (Conductor) o;
        return Objects.equals(nombre, conductor.nombre) && Objects.equals(dni, conductor.dni);
    }

    @Override
    public int hashCode() {
        return Objects.hash(nombre, dni);
    }
}

File: aparcamiento/IvanRC-aparcamiento/IvanRC-aparcamiento-Kotlin/src/main/kotlin/ModelsAparcamiento/Conductor.kt
package ModelsAparcamiento

data class Conductor(val nombre: String, val dni: String) {
    val id: Int = nextId()

    companion object {
        private var contador = 1
        fun nextId(): Int {
            return contador++
        }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Conductor

        if (nombre != other.nombre) return false
        if (dni != other.dni) return false

        return true
    }

    override fun hashCode(): Int {
        var result = nombre.hashCode()
        result = 31 * result + dni.hashCode()
        return result
    }

    override fun toString(): String {
        return "Conductor(nombre='$nombre', dni='$dni', id=$id)"
    }
}